lms/djangoapps/discussion/rest_api/api.py

```diff
@@ -967,20 +970,29 @@ def get_learner_active_thread_list(request, course_key, query_params):
     else:
         comment_client_user = comment_client.User(id=user_id, course_id=course_key, group_id=group_id)
 
-    threads, page, num_pages = comment_client_user.active_threads(query_params)
-    threads = set_attribute(threads, "pinned", False)
-    results = _serialize_discussion_entities(
-        request, context, threads, {'profile_image'}, DiscussionEntity.thread
-    )
-    paginator = DiscussionAPIPagination(
-        request,
-        page,
-        num_pages,
-        len(threads)
-    )
-    return paginator.get_paginated_response({
-        "results": results,
-    })
+    try:
+        threads, page, num_pages = comment_client_user.active_threads(query_params)
+        threads = set_attribute(threads, "pinned", False)
+        results = _serialize_discussion_entities(
+            request, context, threads, {'profile_image'}, DiscussionEntity.thread
+        )
+        paginator = DiscussionAPIPagination(
+            request,
+            page,
+            num_pages,
+            len(threads)
+        )
+        return paginator.get_paginated_response({
+            "results": results,
+        })
+    except CommentClient500Error:
```

Comment:
Why does this error occur?

Reply:
If the selected user has no record in `cs_comment_service`  it will cause a `CommentClient500Error`. This is due to we are now directly querying users from enrollments whether users have any discussions records or not.